### PR TITLE
[FIX] seqan3::detail::get_or_ignore should accept all tuple-like types

### PR DESCRIPTION
**Pull request by marehr**
The old implementation of seqan3::detail::get_or_ignore seemed to work for
many tuple types, but failed for tuple types that have less template
parameters than tuple elements.

### CODE REVIEW

File: include/seqan3/io/detail/record.hpp [hunk 1]
```diff
@@ -137,20 +137,26 @@ auto const & get_or_ignore(record<field_types, field_ids> const & r)
 }
 
 //!\copydoc seqan3::detail::get_or_ignore
-template <size_t i, template <tuple_like ...types_> typename tuple_like_t, typename ...types>
-auto & get_or_ignore(tuple_like_t<types...> & t)
+template <size_t i, typename tuple_t>
+//!\cond
+    requires tuple_like<tuple_t>
+//!\endcond
```
Comment (joergi-w) [inline include/seqan3/io/detail/record.hpp:143]: The following would be shorter and does not need the `cond` part – is there a reason to not write it this way?
```suggestion
template <size_t i, tuple_like tuple_t>
```

**Review by joergi-w — CHANGES_REQUESTED** [review]
I would rename the `test/unit/io/detail/detail_record_test.cpp` test file to something that contains tuple, because it is basically about it. The repetition of detail is also not meaningful. How about `test/unit/io/detail/record_tuple_test.cpp`?

File: include/seqan3/io/detail/record.hpp [hunk 2]
```diff
@@ -137,20 +137,26 @@ auto const & get_or_ignore(record<field_types, field_ids> const & r)
...
         return std::ignore;
 }
 
 //!\copydoc seqan3::detail::get_or_ignore
-template <size_t i, template <tuple_like ...types_> typename tuple_like_t, typename ...types>
-auto const & get_or_ignore(tuple_like_t<types...> const & t)
+template <size_t i, typename tuple_t>
+//!\cond
+    requires tuple_like<tuple_t>
+//!\endcond
```
Comment (marehr) [inline include/seqan3/io/detail/record.hpp:156]:
```suggestion
template <size_t i, tuple_like tuple_t>
```

**Review by joergi-w — APPROVED** [review]
Thank you!

**Review by eseiler — APPROVED** [review]

File: test/unit/io/detail/detail_record_test.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+{};
+
+template <size_t index, typename id_t>
+struct tuple_element<index, custom_tuple<id_t>> : public tuple_element<index, typename custom_tuple<id_t>::base_t>
+{};
+} // namespace std
+
+TEST(get_or_ignore, custom_tuple)
+{
+    custom_tuple<std::string> tuple{};
```
Comment (eseiler) [inline test/unit/io/detail/detail_record_test.cpp:98]: This is what did not work before

**Comment by marehr** [comment]
> I would rename the `test/unit/io/detail/detail_record_test.cpp` test file to something that contains tuple, because it is basically about it. The repetition of detail is also not meaningful. How about `test/unit/io/detail/record_tuple_test.cpp`?

Sorry, I saw your comment and read it, but forgot to answer.  The main problem is that we already have a test that is called `record_test` and it is not possible to have multiple cmake targets with the same name. I have chosen `detail_record_test.hpp`, because the test is in `detail/record_test.hpp` and other test case that have name-clashes prefix one folder up.

I opened an issue https://github.com/seqan/product_backlog/issues/257 (section: Problem with unique test target names) that discusses possible "naming schemes" for tests with name clashes.

**Pull request merged**